### DISCUSSION TITLE
eslint shared tsconfig.json tsconfigRootDir

### DISCUSSION THREAD

**Discussion by lublak** [eslint/eslint#16204] [Q&A] [answered]
In my config there is:

```js
parser: '@typescript-eslint/parser',
parserOptions: {
    project: './tsconfig.json',
    tsconfigRootDir: __dirname,
    sourceType: 'module',
    extraFileExtensions: ['.json'],
}
```

I get: `Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser`

So if i remove `tsconfigRootDir: __dirname,`:

```js
parser: '@typescript-eslint/parser',
parserOptions: {
    project: './tsconfig.json',
    sourceType: 'module',
    extraFileExtensions: ['.json'],
}
```

and move the tsconfig to the root projet it works all fine.
I think somehow the tsconfog not correctly loaded if its not in the root project?
I also tried:
```js
parser: '@typescript-eslint/parser',
parserOptions: {
    project: path.resolve(__dirname, './tsconfig.json'),
    sourceType: 'module',
    extraFileExtensions: ['.json'],
}
```
But this search the tsconfig in the root dir (it ignores the absolute path).
How can i fix it?

**Comment by nzakas** [comment, marked as answer]
This is actually an issue with [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint) rather than ESLint itself. Please open an issue there.

**Reply by lublak** [reply]
@nzakas thanks for the answer :)
